itest/rust/src/object_test.rs

```diff
@@ -856,3 +862,69 @@ fn double_use_reference() {
     double_use.free();
     emitter.free();
 }
+
+#[derive(GodotClass)]
+#[class(init, base=Object)]
+struct SignalReceiver {
+    used: [Cell<bool>; 3],
+    #[base]
+    base: Base<Object>,
+}
+
+#[godot_api]
+impl SignalReceiver {
+    #[func]
+    fn receive_0_arg(&self) {
+        self.used[0].set(true);
+    }
+    #[func]
+    fn receive_1_arg(&self, arg1: i64) {
+        self.used[1].set(true);
+        assert_eq!(arg1, 987);
+    }
+    #[func]
+    fn receive_2_arg(&self, arg1: Gd<Object>, arg2: GodotString) {
+        self.used[2].set(true);
+        assert_eq!(self.base.share(), arg1);
+        assert_eq!(SIGNAL_ARG_STRING, arg2.to_string());
+    }
+}
+
+const SIGNAL_ARG_STRING: &str = "Signal string arg";
+
+#[itest]
+/// Test that godot can call a method that is connect with a signal
+fn signals() {
+    let emitter: Gd<SignalEmitter> = Gd::new_default();
+    let receiver: Gd<SignalReceiver> = Gd::new_default();
+
+    let args = [
+        vec![],
+        vec![Variant::from(987)],
+        vec![
+            Variant::from(receiver.share()),
+            Variant::from(SIGNAL_ARG_STRING),
+        ],
+    ];
+
+    for (i, arg) in args.iter().enumerate() {
+        let signal_name = format!("signal_{}_arg", i);
+        let receiver_name = format!("receive_{}_arg", i);
+
+        emitter.share().upcast::<Object>().connect(
+            signal_name.clone().into(),
+            receiver.callable(receiver_name),
+            0,
+        );
+
+        emitter
+            .share()
+            .upcast::<Object>()
+            .emit_signal(signal_name.into(), arg);
```

Comment:
```suggestion
        emitter.base.emit_signal(signal_name.into(), arg);
```
It's in the same file and for a test, so encapsulation is not the highest priority.